File: server/Services/CryptidEncountersService.cs



namespace cryptipedia.Services;

public class CryptidEncountersService
{
  private readonly CryptidEncountersRepository _repository;

  public CryptidEncountersService(CryptidEncountersRepository repository)
  {
    _repository = repository;
  }

  internal CryptidEncounterProfile CreateCryptidEncounter(CryptidEncounter cryptidEncounterData)
  {
    CryptidEncounterProfile cryptidEncounter = _repository.CreateCryptidEncounter(cryptidEncounterData);
    return cryptidEncounter;
  }

  internal List<EncounteredCryptid> GetCryptidEncountersByAccountId(string accountId)
  {
    List<EncounteredCryptid> cryptids = _repository.GetCryptidEncountersByAccountId(accountId);
    return cryptids;
  }

  internal List<CryptidEncounterProfile> GetCryptidEncountersByCryptidId(int cryptidId)
  {
    List<CryptidEncounterProfile> cryptidEncounterProfiles = _repository.GetCryptidEncountersByCryptidId(cryptidId);
    return cryptidEncounterProfiles;
  }
}

File: client/src/services/CryptidEncountersService.js
import { logger } from "@/utils/Logger.js"
import { api } from "./AxiosService.js"
import { AppState } from "@/AppState.js"
import { CryptidEncounterProfile } from "@/models/CryptidEncounterProfile.js"
import { EncounteredCryptid } from "@/models/EncounteredCryptid.js"

class CryptidEncountersService {
  async deleteCryptidEncounter(cryptidEncounterId) {
    const response = await api.delete(`api/cryptidEncounters/${cryptidEncounterId}`)
    logger.log('DELETED ENCOUNTER', response.data)
    const index = AppState.encounteredCryptids.findIndex(cryptid => cryptid.cryptidEncounterId == cryptidEncounterId)
    AppState.encounteredCryptids.splice(index, 1)
  }
  async getMyEncounteredCryptids() {
    const response = await api.get('account/cryptidEncounters')
    logger.log('GOT MY ENCOUNTERED CRYPTIDS', response.data)
    AppState.encounteredCryptids = response.data.map(pojo => new EncounteredCryptid(pojo))
  }
  async createCryptidEncounter(cryptidEncounterData) {
    const response = await api.post('api/cryptidEncounters', cryptidEncounterData)
    logger.log('CREATED CRYPTID ENCOUNTER 🥸+🦟', response.data)
    AppState.cryptidEncounterProfiles.push(new CryptidEncounterProfile(response.data))
  }
  async getCryptidEncountersByCryptidId(cryptidId) {
    const response = await api.get(`api/cryptids/${cryptidId}/cryptidEncounters`)
    logger.log('GOT CRYPTID ENCOUNTERS 🥸🥸🥸', response.data)
    AppState.cryptidEncounterProfiles = response.data.map(pojo => new CryptidEncounterProfile(pojo))
  }
}

export const cryptidEncountersService = new CryptidEncountersService()